src/marten/db/query/sql/query.cr

```diff
@@ -86,8 +86,8 @@ module Marten
             @using.nil? ? Model.connection : Connection.get(@using.not_nil!)
           end
 
-          def count
-            sql, parameters = build_count_query
+          def count(column_name : String | Symbol | Nil = nil)
+            sql, parameters = build_count_query(column_name)
```

Comment:
We are missing a step here. Indeed, before generating the count SQL query we must first ensure that the "column name" (which is actually a model field name) corresponds to an existing field in the queried model.

In order to do so, we can do something like that:

```crystal
def count(field : String | Nil = nil)
  column_name = if !field.nil?
    field_path = verify_field(raw_field)
    relation_field_path = field_path.select { |field, _r| field.relation? }
    
    if relation_field_path.empty? || field_path.size == 1
      "#{Model.db_table}.#{field_path.first[0].db_column!}"
    else
      join = ensure_join_for_field_path(relation_field_path)
      join.not_nil!.column_name(field_path.last[0].db_column!)
    end
  end

  sql, parameters = build_count_query(column_name)
  
  # ...
```

This will essentially ensure that we:

1. Verify that the field exists (by calling the `#verify_field` method)
2. Use the corresponding DB column name (field names and DB column names do not always match!)
3. Set up a join if the targeted column is part of a relation

Note that `#verify_field` will automatically raise the right exception if the specified field does not exist.